pkg/urbit/vere/io/term.c

```diff
@@ -464,37 +445,63 @@ _term_it_show_cursor(u3_utty* uty_u, c3_w cur_w)
   uty_u->tat_u.mir.cus_w = cur_w;
 }
 
-/* _term_it_show_line(): set current line
+/* _term_it_show_line(): render current line.
 */
 static void
-_term_it_show_line(u3_utty* uty_u, c3_w* lin_w, c3_w len_w, c3_w sap_w)
+_term_it_show_line(u3_utty* uty_u, c3_w wor_w, c3_w sap_w)
 {
-  _term_it_show_wide(uty_u, len_w, lin_w);
+  u3_utat* tat_u = &uty_u->tat_u;
 
-  if ( lin_w != uty_u->tat_u.mir.lin_w ) {
-    if ( uty_u->tat_u.mir.lin_w ) {
-      c3_free(uty_u->tat_u.mir.lin_w);
-    }
-    uty_u->tat_u.mir.lin_w = lin_w;
-  }
-  uty_u->tat_u.mir.len_w = len_w;
-  uty_u->tat_u.mir.sap_w = sap_w;
+  _term_it_dump(uty_u, tat_u->mir.byt_w, tat_u->mir.lin_y);
+
+  //  XX refactor to avoid updating state
+  //
+  tat_u->mir.cus_w += wor_w;
+  tat_u->mir.wor_w = wor_w;
+  tat_u->mir.sap_w = sap_w;
```

Comment:
Not sure what's going on here. Don't you only need to update these alongside `lin_y` (as done in `_term_it_set_line`)? If `lin_y` doesn't change, why would these? Guessing there's some relation to the bytes<->words measurements I'm not fully grokking.

I guess we still update `cus_w` regardless because rendering puts our cursor at the end of whatever we rendered...

Reply:
We currently have to (re-)set these values, as some sequences of output operations change them (ie, `_show_clear()` followed by `_show_line()`).